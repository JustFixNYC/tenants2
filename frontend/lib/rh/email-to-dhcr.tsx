import React from "react";

import {
  EmailSubject,
  asEmailStaticPage,
} from "../static-page/email-static-page";
import {
  getBoroughChoiceLabels,
  BoroughChoice,
} from "../../../common-data/borough-choices";
import { Trans, t } from "@lingui/macro";
import { li18n } from "../i18n-lingui";
import { TransformSession } from "../util/transform-session";
import { AllSessionInfo } from "../queries/AllSessionInfo";

function getEmailInfo(s: AllSessionInfo) {
  const rh = s.rentalHistoryInfo;

  if (!rh) return null;

  const { apartmentNumber } = rh;
  const fullLegalName = `${rh.firstName} ${rh.lastName}`;
  const borough = getBoroughChoiceLabels()[rh.borough as BoroughChoice];
  const fullAddress = `${rh.address}, ${borough} ${rh.zipcode}`.trim();

  return { fullLegalName, fullAddress, apartmentNumber };
}

export const RhEmailToDhcr: React.FC<{}> = () => {
  return (
    <TransformSession transformer={getEmailInfo}>
      {(i) => (
        <>
          <EmailSubject value={li18n._(t`Request for Rent History`)} />

          <Trans id="justfix.rhRequestToDhcr">
            <p>DHCR administrator,</p>
            <p>
              I, {i.fullLegalName}, am currently living at {i.fullAddress} in
              apartment {i.apartmentNumber}, and would like to request the
              complete Rent History for this apartment back to the year 1984.
            </p>
            <p>
              Thank you,
              <br />
              {i.fullLegalName}
            </p>
          </Trans>
        </>
      )}
    </TransformSession>
  );
};

export const RhEmailToDhcrStaticPage = asEmailStaticPage(RhEmailToDhcr);
