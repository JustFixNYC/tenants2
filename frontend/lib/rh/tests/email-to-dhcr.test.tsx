import React from "react";
import { AppTesterPal } from "../../tests/app-tester-pal";
import { RhEmailToDhcr } from "../email-to-dhcr";

test("email to dhcr renders", () => {
  const pal = new AppTesterPal(<RhEmailToDhcr />, {
    session: {
      rentalHistoryInfo: {
        firstName: "boop",
        lastName: "jones",
        address: "150 DOOMBRINGER STREET",
        apartmentNumber: "2",
        phoneNumber: "2120000000",
        borough: "MANHATTAN",
        zipcode: "10001",
        addressVerified: true,
      },
    },
  });

  expect(pal.rr.container).toMatchSnapshot();
});
