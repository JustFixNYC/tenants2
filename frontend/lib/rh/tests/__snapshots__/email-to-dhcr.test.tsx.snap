// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`email to dhcr renders 1`] = `
<div>
  <p
    class="jf-email-subject"
  >
    Subject: 
    Request for Rent History
  </p>
  <p>
    DHCR administrator,
  </p>
  <p>
    I, boop jones, am currently living at 150 DOOMBRINGER STREET, Manhattan 10001 in apartment 2, and would like to request the complete Rent History for this apartment back to the year 1984.
  </p>
  <p>
    Thank you,
    <br />
    boop jones
  </p>
</div>
`;
